File: programs/speed_comparison/python/speed.py
import timeit

def func():
    pi = 1.0
    sign = -1
    for i in range(1,10000000):
        pi   = pi + sign / (2 * i + 1)
        sign = sign * -1
    return pi

if __name__ == '__main__':
    pi = func();
    for i in range(10):
        print("time: ", timeit.timeit("func()", setup="from __main__ import func", number=1))
    print("result: ", pi * 4)

File: programs/speed_comparison/julia/speed.jl
function speed_test() 
    pi = 0
    for i = 0:10000000
        pi += (-1.0)^i / (2.0 * i + 1)
    end
    return 4*pi
end

#@time speed_test()
#@time speed_test()
for n = 1:100
    println(@time speed_test())
end
#@time speed_test()